router/http/src/main/scala/io/buoyant/router/http/StripHopByHopHeadersFilter.scala

```diff
@@ -0,0 +1,55 @@
+package io.buoyant.router.http
+
+import com.twitter.finagle.http.Fields._
+import com.twitter.finagle.http.{Message, Request, Response}
+import com.twitter.finagle.{Service, ServiceFactory, SimpleFilter, Stack}
+
+object StripHopByHopHeadersFilter {
+
+  object HopByHopHeaders {
+    val Headers = Seq(
+      Connection,
+      ProxyAuthenticate,
+      ProxyAuthorization,
+      Te,
+      Trailer,
+      TransferEncoding,
+      Upgrade
+    )
+
+    def scrub(msg: Message): Unit = {
+      Headers.foreach {
+        case h@Connection =>
+          val headersListedInConnection: Seq[String] = msg.headerMap.remove(h) match {
+            case Some(s) => s.split(",").map(_.trim).filter(_.nonEmpty)
+            case None => Nil
+          }
+          headersListedInConnection.foreach(msg.headerMap.remove(_))
+        case h => msg.headerMap.remove(h)
+      }
+    }
+  }
```

Comment:
D'oh, of course you're right.  I'd be inclined to remove the foreach loop and just enumerate the headers in the function.